### ISSUE TITLE
Task is showing dependency with "task list" report but not on "task next" report

### ISSUE THREAD

**Issue by sdondley** [GothenburgBitFactory/taskwarrior#2151]
Thanks for this great piece of software!

I'm new to TW. I'm not sure if this is a bug or if I'm just confused.

I'm trying to figure out why a particular task, with a Task ID: 1, has a "D" in the "dependency" column with the "task list" command:

```
ID Age   D Project                   Due        Description                                                              Urg
 3 3h      work.clients.ufcw1        2019-04-25 Enable two-factor authentication on Apple Developer account              10.1
 5 2h      work.clients.ilwu63       2019-04-26 Have Don take a look at issue                                            17.6
 6 2h      work.clients.ilwu63       2019-04-26 Delete old files                                                         17.6
12 1h      work:clients:ibew99       2019-04-29 Make sure Don adds email to spam filter                                  8.27
 2 7h      blog                                 Play around with Docker taskserver container to do research for blog        1
 4 2h    D iluw63                               Fix issue with uploading documents                                         -4
 1 12h   D personal                             Campaign finance report                                                     1
```

As far as I can tell, the task has no dependency:

- When I do `task 1` it doesn't show any dependencies:

```
Name          Value
ID            1
Description   Campaign finance report
Status        Pending
Project       personal
Entered       2019-04-25 02:48:52 (12h)
Last modified 2019-04-25 14:50:01 (42min)
Virtual tags  PENDING READY UNBLOCKED PROJECT
UUID          c38efc84-5749-4ecd-8827-7372d96a6606
```

- When I do `task next` it does not show any dependencies in the "Deps" column:

```
ID Age   Deps Project                   Due  Description                                                          Urg
 5 2h         work.clients.ilwu63       8h   Have Don take a look at issue                                        17.6
 6 2h         work.clients.ilwu63       8h   Delete old files                                                     17.6
 3 3h         work.clients.ufcw1        -15h Enable two-factor authentication on Apple Developer account          10.1
12 1h         work:clients:ibew99       3d   Make sure Don adds email to spam filter                              8.27
 1 12h        personal                       Campaign finance report                                                 1
 2 6h         blog                           Play around with Docker taskserver container to do research for blog    1
 7 1h         personal.tech.vim              How do you just forward to next spelling error?                         1
 8 1h         personal.tech.vim              How do you just forward to previous spelling error?                     1
 9 1h         personal.tech.vim              What's the quickest way to correct a spelling mistake you just made?    1
10 1h         personal.tech.taskwarrior      Check out vim-taskwarrior                                               1
11 1h         personal.tech                  is collaboration possible with TW? If so, how?                          1
13 1h         personal.tech.taskwarrior      What's a good way to integrate TW with gmail?                           1
14 42min      personal                       Get car re-inspected                                                    1
 4 2h    5 6  iluw63                         Fix issue with uploading documents                                     -4
```

Thanks to anyone who can offer help.

**Comment by pbeckingham** [comment]
Task 1 does not have any dependencies.  Show us the output for ’task list’, where you say that it does.

Paul

**Comment by sdondley** [comment]
It's the first set of output. I edited the file after posting because I forgot to put it in.

**Comment by sdondley** [comment]
So what do you think, @pbeckingham? Bug or no?

**Comment by pbeckingham** [comment]
Too early to tell.  Can you please run this, so I can test locally:

`$ task 1,2,4 export`

I'm interested in the data.

**Comment by sdondley** [comment]
```
[
{"id":1,"depends":"d0eeffe7-1ce1-4e05-851f-6a8172debcef","description":"Campaign finance report","entry":"20190425T064852Z","modified":"20190425T185001Z","project":"personal","status":"pending","uuid":"c38efc84-5749-4ecd-8827-7372d96a6606","urgency":1},
{"id":2,"description":"Play around with Docker taskserver container to do research for blog","entry":"20190425T123521Z","modified":"20190425T124019Z","project":"blog","status":"pending","uuid":"021f2723-f487-4991-aabd-f60351f5b71f","urgency":1},
{"id":4,"depends":"a3f1eba4-f12e-4aa2-9ad7-746f51248793,0bc4b198-2d90-4438-9422-9b05e541ad90","description":"Fix issue with uploading documents","entry":"20190425T164619Z","modified":"20190425T165120Z","project":"iluw63","status":"pending","uuid":"6d2f987a-77f1-4854-93e1-700196e7a02c","urgency":-4}
]
```

**Comment by pbeckingham** [comment]
Thank you for the data.  I think it's a bug, but it's debatable.

The `next` report includes the column `depends`.  This enumerates only the blocking dependencies.

The `list` report include the column `depends.indicator`, which is the same data, but shows a one character indicator.  This shows the `D` for any task that has dependencies, blocking or not.

**Comment by tbabej** [comment]
Do I understand correctly that the bug here is to unify the behaviour of `depends` and `depends.indicator` columns?

**Comment by drehak** [comment]
@tbabej I've investigated this bug and it looks like it got fixed between v2.5.3 and v.2.6.0, specifically in commit 413b8d2.

### Reproduction

The tasks on which I'm testing this:

```
Name          Value                                  
ID            7                                      
Description   Lil' Blocked
Status        Pending                                
Project       block
Entered       2022-04-25 21:49:12 (1h)               
Last modified 2022-04-25 22:42:27 (12min)
Virtual tags  LATEST PENDING PROJECT READY UNBLOCKED 
UUID          18d3e87e-8f23-40ad-9b8a-827f58894b0f
Urgency       1                                      

    project      1 *    1 =      1
                            ------
                                 1

Date                Modification                                               
2022-04-25 21:50:10 Dependency on '03ffc467-5faa-4d78-b1ba-63cf06cb61f8' added.
2022-04-25 22:42:27 Project set to 'block'.


Name                  Value                               
ID                    -                                   
Description           Lil' Blocker
Status                Completed                           
Project               block
This task is blocking 7 Lil' Blocked                      
Entered               2022-04-25 21:49:13 (1h)
End                   2022-04-25 21:50:29                 
Last modified         2022-04-25 22:42:27 (12min)
Virtual tags          COMPLETED PROJECT UNBLOCKED         
UUID                  03ffc467-5faa-4d78-b1ba-63cf06cb61f8
Urgency               1                                   

    project      1 *    1 =      1
                            ------
                                 1

Date                Modification                                 
2022-04-25 21:50:29 End set to '2022-04-25 21:50:29'.            
                    Status changed from 'pending' to 'completed'.
2022-04-25 22:42:27 Project set to 'block'.
```

Output for commit 413b8d2:

```
$ task list proj:block

ID Age  Project Description      Urg 
 7 1h   block   Lil' Blocked        1

1 task
Filter: ( status = pending and tags _notag_ WAITING ) and ( proj = block )
$ task next proj:block

ID Age  Project Description  Urg 
 7 1h   block   Lil' Blocked    1

1 task
Filter: ( status = pending and tags _notag_ WAITING and limit:page ) and ( proj = block )
```

Output for commit 9e67f4f (parent of 413b8d2):

```
$ task list proj:block

ID Age  D Project Description      Urg 
 7 1h   D block   Lil' Blocked        1

1 task
Filter: ( status = pending and tags _notag_ WAITING ) and ( proj = block )
$ task next proj:block

ID Age  Project Description  Urg 
 7 1h   block   Lil' Blocked    1

1 task
Filter: ( status = pending and tags _notag_ WAITING and limit:page ) and ( proj = block )
```

The behavior between the colums `depends` and `depends.indicator` seems to have unified, so I guess you can close this.

**Comment by tbabej** [comment]
Thanks for digging in! This indeed looks like it got inadvertently fixed.
